lints/lint_idn_dnsname_must_be_nfkc.go

```diff
@@ -0,0 +1,51 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"golang.org/x/net/idna"
+	"golang.org/x/text/unicode/norm"
+	"strings"
+)
+
+type IDNNotNFKC struct {
+	// Internal data here
+}
+
+func (l *IDNNotNFKC) Initialize() error {
+	return nil
+}
+
+func (l *IDNNotNFKC) CheckApplies(c *x509.Certificate) bool {
+	return util.IsExtInCert(c, util.SubjectAlternateNameOID)
+}
+
+func (l *IDNNotNFKC) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	for _, dns := range c.DNSNames {
+		splitLabels := strings.Split(dns, ".")
+		for _, label := range splitLabels {
+			if strings.HasPrefix(label, "xn--") {
+				//IDN domain name
+				unicodeLabel, err := idna.ToUnicode(label)
+				if err != nil {
+					return ResultStruct{Result: Fatal}, nil
```

Comment:
Should this be `Fatal` or `Error`? If it's not unicode it's not valid NFKC.

Reply:
We've been treating external library calls that error as `Fatal`, so we don't confuse them with an actual `Error` from the lint.

Reply:
This means that an invalid IDN label will Fatal this check, is that intended?

Reply:
I think that's the right call. What I can do is add another lint that returns a warn if there's an invalid IDN label. Does that sound reasonable?

Reply:
The intention is to warn in that case, so should this Pass? (or whatever the value is if the check wasn't run)

Reply:
Sorry, I'm a bit confused - if `idna.ToUnicode` returns an error, do you think this should instead be a Pass, rather than a Fatal?

Reply:
If there is a separate rule that warns in that case, it seems like this rule should not return a fatal, in the same that a rule covering an optional extension works when the extension doesn't exist.

Reply:
OH, I see. Rather than `Fatal`, return `NA`, as in, this rule doesn't apply here. Makes sense to me 👍